magma/array.py

```diff
@@ -342,7 +342,7 @@ def value(self):
         if self.iswhole(ts):
             return ts[0].name.array
 
-        return type(self)(*ts)
+        return type(self).flip()(*ts)
```

Comment:
is this a separate bug?

Reply:
This bug was revealed when enforcing this directionality changed.

Basically, when you have an input, you can retrieve what was driving it by calling `.value`.  However, the driver is an output, so you should be creating the flipped type.  It turns out the code using these values was not affected by the directionality (it's used by the backend to find out what is driving a value), but this is technically what should be happening.